lib/matplotlib/axes/_axes.py

```diff
@@ -2320,7 +2320,59 @@ def _convert_dx(dx, x0, xconv, convert):
             # we do by default and convert dx by itself.
             dx = convert(dx)
         return dx
+    
+    @staticmethod
+    def _parse_bar_color_args(kwargs, get_next_color_func):
+        """
+        Helper function to process color-related arguments of `.Axes.bar`.
+
+        Argument precedence for facecolors:
+
+        - kwargs['facecolor']
+        - kwargs['color']
+        - 'b' if in classic mode else the result of ``get_next_color_func()``
+
+        Argument precedence for edgecolors:
 
+        - kwargs['edgecolor']
+        - kwargs['color']
+        - None
+
+        Parameters
+        ----------
+        kwargs : dict
+            Additional kwargs. If these keys exist, we pop and process them:
+            'facecolor', 'edgecolor', 'color'
+            Note: The dict is modified by this function.
+        get_next_color_func : callable
+            A callable that returns a color. This color is used as facecolor
+            if no other color is provided.
+
+        Returns
+        -------
+        facecolor
+            The facecolor.
+        edgecolor
+            The edgecolor.
+        """
+        color = kwargs.pop('color', None)
+
+        facecolor = kwargs.pop('facecolor', color)
+        edgecolor = kwargs.pop('edgecolor', color)
+
+        facecolor = (facecolor if facecolor is not None
+                 else "b" if mpl.rcParams['_internal.classic_mode']
```

Comment:
I guess you made "b" the default in classic mode for consistency with `scatter`.  However this is not the established behaviour for `bar` and setting it to blue is likely the cause of at least some of your test failures.  For `scatter` it will only be there for consistency with old Matplotlib versions.  I think we should just use the `get_next_color_func` here.